addons/pos_self_order/controllers/kiosk_entry.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+import werkzeug
+
+from odoo import http
+from odoo.http import request
+from odoo.addons.pos_self_order.controllers.utils import reduce_privilege
+from odoo.tools.misc import get_lang
+
+
+class PosSelfKiosk(http.Controller):
+    @http.route(["/kiosk/<config_id>", "/kiosk/<config_id>/<path:subpath>"], auth="public", website=True, sitemap=True)
```

Comment:
Because we have a router in the frontend that changes the URL.

Sometimes it can be `http://localhost:4444/kiosk/5/product/42` and without the subpath, the backend route won't understand that we want to be redirected to the kiosk.